trackpy/linking/partial.py

```diff
@@ -0,0 +1,155 @@
+        Default is ['y', 'x'], or ['z', 'y', 'x'] when 'z' is present in f
+    t_column : str, optional
+        Default is 'frame'
+    predictor : function, optional
+        Improve performance by guessing where a particle will be in
+        the next frame.
+        For examples of how this works, see the "predict" module.
+    adaptive_stop : float, optional
+        If not None, when encountering an oversize subnet, retry by progressively
+        reducing search_range until the subnet is solvable. If search_range
+        becomes <= adaptive_stop, give up and raise a SubnetOversizeException.
+    adaptive_step : float, optional
+        Reduce search_range by multiplying it by this factor.
+    link_strategy : {'recursive', 'nonrecursive', 'numba', 'drop', 'auto'}
+        algorithm used to resolve subnetworks of nearby particles
+        'auto' uses numba if available
+        'drop' causes particles in subnetworks to go unlinked
+
+    Returns
+    -------
+    DataFrame with added column 'particle' containing trajectory labels.
+    The t_column (by default: 'frame') will be coerced to integer."""
+    if pos_columns is None:
+        pos_columns = guess_pos_columns(f)
+    ndim = len(pos_columns)
+    search_range = validate_tuple(search_range, ndim)
+    if kwargs.get('memory', 0) > 0:
+        warnings.warn("Particles are not memorized over patch edges.")
+
+    full_range = (int(f[t_column].min()), int(f[t_column].max()) + 1)
+    start, stop = link_range
+    assert start < stop
+    if start is None:
+        start = full_range[0]
+    elif start < full_range[0]:
+        start = full_range[0]
+
+    if stop is None:
+        stop = full_range[1]
+    elif stop > full_range[1]:
+        stop = full_range[1]
+
+    link_frame_nos = range(start, stop)
+
+    # copy the dataframe
+    f = f.copy()
+    # coerce t_column to integer type
+    if not np.issubdtype(f[t_column].dtype, np.integer):
+        f[t_column] = f[t_column].astype(np.integer)
+    # sort on the t_column
+    pandas_sort(f, t_column, inplace=True)
+
+    if 'particle' in f and (start > full_range[0] or stop < full_range[1]):
+        f['_old_particle'] = f['particle'].copy()
+    elif not 'particle' in f:
+        f['particle'] = -1
+
+    coords_iter = coords_from_df_partial(f, pos_columns, t_column,
+                                         link_frame_nos)
+    for i, _ids in link_iter(coords_iter, search_range, **kwargs):
+        f.loc[f[t_column] == i, 'particle'] = _ids
+
+    if '_old_particle' in f:
+        reconnect_traj_patch(f, (start, stop), '_old_particle', t_column)
+        f.drop('_old_particle', axis=1, inplace=True)
+
+    return f
+
+
+def reconnect_traj_patch(f, link_range, old_particle_column, t_column='frame'):
+    """Reconnect the trajectory inside a range of frames to the trajectories
+    outside the range. Requires a column with the original particle indices."""
+    # TODO Does not yet work fully in combination with memory
```

Comment:
I would remove this comment since I'm pretty sure this is impossible to solve generally.